lib/interval/interval.ex

```diff
@@ -220,8 +241,12 @@ defmodule Timex.Interval do
       if has_recursion_ended?(current_date, end_date, right_open) do
         {:done, acc}
       else
-        next_date = Timex.shift(current_date, keywords)
-        do_reduce({next_date, end_date, right_open, keywords}, fun.(current_date, acc), fun)
+        case Timex.shift(current_date, keywords) do
+          {:error, _} ->
+            {:halt, acc}
+          next_date ->
+            do_reduce({next_date, end_date, right_open, keywords}, fun.(current_date, acc), fun)
+        end
```

Comment:
Even though `Timex.new/1` and `Timex.with_step/2` now return errors when given an invalid step, we still need to prevent infinite recursion here since you can just create an Interval struct by hand if you wanted.